### PR TITLE
[ENHANCEMENT] CLI `docs clean` command implemented for v3 api

### PR DESCRIPTION
**Pull request by Aylr**
[ENHANCEMENT] CLI `docs clean` command implemented for v3 api

### CODE REVIEW

File: great_expectations/cli/docs.py [hunk 1]
```diff
@@ -84,35 +84,26 @@ def docs_list(ctx):
 @click.option(
     "--all",
     "-a",
+    "all_sites",
```
Comment (anthonyburdi) [inline great_expectations/cli/docs.py:87]: YES!! I love when the thing says what it does.

File: great_expectations/cli/docs.py [hunk 2]
```diff
@@ -84,35 +84,26 @@ def docs_list(ctx):
...
-    if failed and context is not None:
-        toolkit.send_usage_message(
-            data_context=context, event="cli.docs.clean", success=False
+    if site_name is None and all_sites is False:
+        toolkit.exit_with_failure_message_and_stats(
+            context,
+            usage_event="cli.docs.clean",
+            message="<red>Please specify --all to remove all sites or specify a specific site using --site_name</red>",
         )
+    context.clean_data_docs(site_name=site_name)
```
Comment (anthonyburdi) [inline great_expectations/cli/docs.py:103]: Interesting - so if site_name is specified then only that site is cleaned, and if no site name is specified and all_sites is True then `context.clean_data_docs()` loops through all of them. This isn't super hard to figure out but it is slightly confusing as to why there is no path for all_sites at first - what do you think about adding a tiny note like this:
```suggestion
    # if site_name is None, context.clean_data_docs(site_name=site_name) will clean all sites.
    context.clean_data_docs(site_name=site_name)
```

File: tests/cli/test_docs.py [hunk 3]
```diff
@@ -296,3 +296,123 @@ def test_docs_list(
...
+    obs_urls = context.get_docs_sites_urls()
+    assert len(obs_urls) == 1
+    expected_index_path = os.path.join(
+        context.root_directory,
+        context.GE_UNCOMMITTED_DIR,
+        "data_docs",
+        "local_site",
+        "index.html",
+    )
+    assert os.path.isfile(expected_index_path)
```
Comment (anthonyburdi) [inline tests/cli/test_docs.py:314]: I love these asserts in fixtures to guard against our future selves.

File: tests/cli/test_docs.py [hunk 4]
```diff
@@ -296,3 +296,123 @@ def test_docs_list(
...
+
+    assert_no_logging_messages_or_tracebacks(
+        my_caplog=caplog,
+        click_result=result,
+    )
+
+
+@pytest.mark.parametrize(
```
Comment (anthonyburdi) [inline tests/cli/test_docs.py:367]: No way. I am totally going to steal this to refactor the `--assume-yes` `--yes` `-y` triple that I wrote earlier.

File: tests/cli/test_docs.py [hunk 5]
```diff
@@ -296,3 +296,123 @@ def test_docs_list(
...
+        "--v3-api docs clean --all",
+        "--v3-api docs clean -a",
+        "--v3-api docs clean --site-name local_site",
+        "--v3-api docs clean -s local_site",
+    ],
+)
+@mock.patch(
+    "great_expectations.core.usage_statistics.usage_statistics.UsageStatisticsHandler.emit"
+)
+def test_docs_clean_vali_site_name_and_all_flag_combinations_clean_sites(
```
Comment (anthonyburdi) [inline tests/cli/test_docs.py]: I just had to nitpick something 😂 
```suggestion
def test_docs_clean_valid_site_name_and_all_flag_combinations_clean_sites(
```

File: tests/cli/test_docs.py [hunk 6]
```diff
@@ -296,3 +296,123 @@ def test_docs_list(
...
+        "index.html",
+    )
+    assert os.path.isfile(expected_index_path)
+    return context
+
+
+@mock.patch(
+    "great_expectations.core.usage_statistics.usage_statistics.UsageStatisticsHandler.emit"
+)
+def test_docs_clean_no_site_specified_raises_helpful_error(
```
Comment (anthonyburdi) [inline tests/cli/test_docs.py:321]: Maybe one more test for `test_docs_clean_invalid_site_specified_raises_helpful_error`?

**Review by anthonyburdi — APPROVED** [review]
Just some minor suggested changes, non-blocking so feel free to implement or merge anyway if you don't agree.

File: tests/cli/test_docs.py (hunk 5 above)
Comment (Aylr) [reply tests/cli/test_docs.py]: nice catch!

File: tests/cli/test_docs.py (hunk 6 above)
Comment (Aylr) [reply tests/cli/test_docs.py:321]: This functionality doesn't exist in v2 at all. Let me make another task and PR to address this gap.

**Pull request merged**